algorithms/kernel/k_nearest_neighbors/bf_knn_classification_predict_dense_default_batch_fpt_cpu.cpp

```diff
@@ -0,0 +1,52 @@
+/* file: bf_knn_classification_predict_dense_default_batch_fpt_cpu.cpp */
+/*******************************************************************************
+* Copyright 2014-2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of prediction stage of K-Nearest Neighbors algorithm.
+//--
+*/
+
+#include "bf_knn_classification_predict_dense_default_batch.h"
+#include "oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i"
+#include "bf_knn_classification_predict_dense_default_batch_container.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace bf_knn_classification
+{
+namespace prediction
+{
+namespace interface1
+{
+
+template class BatchContainer<DAAL_FPTYPE, defaultDense, DAAL_CPU>;
+
+} // namespace interface1
+
+namespace internal
+{
+
+template class KNNClassificationPredictKernelUCAPI<DAAL_FPTYPE>;
```

Comment:
My recommendation is to divide this file into two parts, each one containing a single explicit instantiation clause. To initialize `BatchContainer`, you do not need to know about the implementation of `KNNClassificationPredictKernelUCAPI` kernel. This can be dangerous. Please, look at how the same kernels are instantiated for other algorithms (covariance, log regression, ...)